NDTensors/src/lib/Sectors/src/sector.jl

```diff
@@ -0,0 +1,128 @@
+
+struct Sector{Data} <: AbstractCategory
+  data::Data
+end
+
+Sector(nt::NamedTuple) = Sector{NamedTuple}(nt_sort(nt))
+
+Sector(; kws...) = Sector((; kws...))
+
+function Sector(pairs::Pair...)
+  keys = ntuple(n -> Symbol(pairs[n][1]), length(pairs))
+  vals = ntuple(n -> pairs[n][2], length(pairs))
+  return Sector(NamedTuple{keys}(vals))
+end
+
+Sector(cats::AbstractCategory...) = Sector([cats...])
```

Comment:
I was picturing we would use a tuple instead of a `Vector` for performance, especially because the `cats` may not all be the same type.